src/combine.cpp

```diff
@@ -1058,10 +1058,7 @@ void do_symbol_check(chunk_t *prev, chunk_t *pc, chunk_t *next)
       || pc->type == CT_STRUCT
       || pc->type == CT_UNION)
    {
-      if (prev->type != CT_TYPEDEF)
```

Comment:
Do you know why this was introduced in the first place?

Reply:
No, this goes as far back as cb24cf1487844dbd63a010571dbb7266c7417b9a, but there are no clues about why this was introduced. According to the commit message this commit fixes a bug, but the change set is rather big so I am not sure what exactly got fixed.